### PR TITLE
Avoid a newline after type parameter attributes

### LINKED ISSUE #38916: Formatting of type parameter with attribute includes a line break
A line break is included in the formatting of a type parameter with an attribute.

Create a .NET Core 3.0 C# project and Go To Definition on `System.IEquatable<T>`:

GoToDefinition

### PR DESCRIPTION
**Pull request by sharwell**
Fixes #38916

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Don't we have metadata-as-source tests?

**Comment by sharwell** [comment]
Yep found them!

File: src/EditorFeatures/Test/MetadataAsSource/MetadataAsSourceTests.cs [hunk 1]
```diff
@@ -2198,8 +2271,7 @@ void M()
 
 using System.Runtime.CompilerServices;
 
-public class [|TestType|]<[NullableAttribute(1)]
-T> where T : notnull
+public class [|TestType|]<[NullableAttribute(1)] T> where T : notnull
```
Comment (jasonmalinowski) [inline src/EditorFeatures/Test/MetadataAsSource/MetadataAsSourceTests.cs]: Do we have a test of this behavior that doesn't involve NullableAttribute? The fact it's currently dumping this in metadata as source is something we plan on fixing, and if updating this test removes the coverage for the bug we'd want to test this some other way.

Comment (sharwell) [reply src/EditorFeatures/Test/MetadataAsSource/MetadataAsSourceTests.cs]: Yes. I added a test that uses a custom attribute specifically for this purpose.

**Pull request merged**